tests/test-cases-lib.sh

```diff
@@ -75,6 +107,61 @@ function TEST_road-traffic-predictor() {
 
 }
 
+function TEST_realtime-trade-monitor() {
+  echo "********** START TEST: realtime-trade-monitor **********"
+
+  local hzVersion=5.0-SNAPSHOT
+  local timeout=200
+  local pause=20
+  local demo_root=$1
+  local log_file=realtime-trade-monitor-log.txt
+
+  echo "---------- realtime-trade-monitor: preparing env - starting kafka service and creating \"trades\" topic"
+  cd $demo_root/tests/tools/qe-kafka-manager
+  timeout $timeout mvn exec:java -DkafkaTopic=trades&
+  local KAFKA_PID=$!
+  sleep $pause
+
+  echo "---------- realtime-trade-monitor: preparing env - starting trades producer"
+  cd $demo_root/realtime-trade-monitor
+  timeout $timeout java -jar trade-producer/target/trade-producer-${hzVersion}.jar 127.0.0.1:9092 5 | tee ${log_file}&
+  local PRODUCER_PID=$!
+  sleep $pause
+
+  echo "---------- realtime-trade-monitor: starting jet server"
+  timeout $timeout java -jar jet-server/target/jet-server-${hzVersion}.jar | tee ${log_file}&
+  local JET_PID=$!
+  sleep $pause
+
+  echo "---------- realtime-trade-monitor: starting jet jobs to ingest and aggregate trades"
+  java -jar trade-queries/target/trade-queries-${hzVersion}.jar load-symbols | tee ${log_file}
+  sleep $pause
+  java -jar trade-queries/target/trade-queries-${hzVersion}.jar ingest-trades 127.0.0.1:9092 | tee ${log_file}
+  sleep $pause
+  java -jar trade-queries/target/trade-queries-${hzVersion}.jar aggregate-query 127.0.0.1:9092 | tee ${log_file}
+  sleep $pause
+
+  echo "********** Launch data-observer to make sure landingMap is being updated by job **********"
+  cd $demo_root/tests/tools/qe-data-observer/target || exit
+  java -jar qe-data-observer-5.0-SNAPSHOT-jar-with-dependencies.jar jet query1_Results
+  local TEST_RESULT=$?
+
+  #  cleaning up the environment - killing all launched processes
+  kill -KILL $KAFKA_PID
```

Comment:
Interesting... Need to investigate a bit

Reply:
$1 returning PID of the last process launched. It could be a maven process or service process in random order. This completely redesigned to catch processes by name and kill both maven and service process.